includes/utilities.inc

```diff
@@ -17,21 +17,38 @@ function islandora_usage_stats_log_object_view(AbstractObject $object) {
     if (($last_viewed + variable_get('islandora_usage_stats_cooldown_seconds', 300)) < REQUEST_TIME) {
       module_load_include('inc', 'islandora_basic_collection', 'includes/utilities');
       module_load_include('inc', 'islandora_usage_stats', 'includes/db');
-      $pid_id = islandora_usage_stats_get_pid_id($object->id);
-      $object_access_id = islandora_usage_stats_object_view_to_db($pid_id);
+      try {
+        $pid_id = islandora_usage_stats_get_pid_id($object->id);
+        $object_access_id = islandora_usage_stats_object_view_to_db($pid_id);
+      }
+      catch (Exception $e) {
+        watchdog('islandora_usage_stats', 'Error: islandora_usage_stats_pid_to_db() could not load %pid', array('%pid' => $object->id), WATCHDOG_ERROR);
```

Comment:
@DiegoPino If it helps any, here's how I have it mapped out:
`islandora_usage_stats_pid_to_db()` is the original object getting passed into `islandora_usage_stats_log_object_view()`

- `islandora_usage_stats_pid_to_db()` in includes/db.inc is where the error comes from, since its trying to `islandora_object_load()` a PID that may not exist. If it can't `islandora_object_load()`, throw an error.
- `islandora_usage_stats_get_pid_id` in includes/db.inc calls `islandora_usage_stats_pid_to_db()` if the PID it is passed doesn't have a database ID yet.
- `islandora_usage_stats_log_object_view()` in includes/utilities.inc calls `islandora_usage_stats_get_pid_id` three times (where the three try/catch blocks are). Of the three calls to `islandora_usage_stats_get_pid_id`:
  - The first is to log the actual object getting viewed, so this should always exist. Its also the object that gets passed to `islandora_usage_stats_log_object_view()`, so it should complain about an invalid object on invocation before ever reaching a call to `islandora_usage_stats_get_pid_id`.
  - The second call to `islandora_usage_stats_get_pid_id` is to log the CModel of the object that was viewed. This is pulled from the RELS-EXT directly, so if the RELS-EXT is wrong (and we can force it to be wrong by adding fake CModels with tuque) the exception is triggered.
  - The third and final call to `islandora_usage_stats_get_pid_id` is to log the parents of the object that was accessed, and this is once again pulled straight from the RELS-EXT directly so if the RELS-EXT is wrong, it will trigger the exception.
  - Of these three calls to `islandora_usage_stats_get_pid_id`, only the last two could possibly call nonexistent objects, and only the last call is likely to happen in practice (IMHO).